packages/flutter/lib/src/painting/cupertino_rounded_rectangle_border.dart

```diff
@@ -0,0 +1,511 @@
+        ..lineTo(originX, originY + 1.52866483 * radius)
+        ..cubicTo(leftX(0), topY(1.63527822),
+            leftX(0), topY(1.29884040),
+            leftX(0.10012126), topY(0.99544257))
+        ..lineTo(leftX(0.11451443), topY(0.93667966))
+        ..cubicTo(leftX(0.31920189), topY(0.37430552),
+            leftX(0.85376549), topY(0),
+            leftX(1.45223153), topY(0))
+        ..cubicTo(centerX, topY(0),
+            centerX, topY(0),
+            centerX, topY(0))
+        ..lineTo(centerX, topY(0))
+        ..close();
+    }
+
+    Path roundedRect3a () {
+      return Path()
+        ..moveTo(centerX, topMidY(0))
+        ..lineTo(centerX, topMidY(0))
+        ..cubicTo(centerX, topMidY(0),
+            centerX, topMidY(0),
+            centerX, topMidY(0))
+        ..lineTo(centerX, topMidY(0))
+        ..cubicTo(rightX(0.68440646), topY(0.00000001),
+            rightX(0), topY(0.68440658),
+            rightX(0), topY(1.52866483))
+        ..cubicTo(rightX(0), topY(1.52866495),
+            rightX(0), topY(1.52866495),
+            rightX(0), topY(1.52866507))
+        ..cubicTo(rightX(0), topY(1.52866483),
+            rightX(0), topY(1.52866483),
+            rightX(0), topY(1.52866483))
+        ..lineTo(rightMidX(0), centerY)
+        ..cubicTo(rightX(0), bottomY(1.52866471),
+            rightX(0), bottomY(1.52866471),
+            rightX(0), bottomY(1.52866471))
+        ..lineTo(rightX(0), bottomY(1.52866471))
+        ..cubicTo(rightX(0), bottomY(0.68440646),
+            rightX(0.68440646), bottomY(0),
+            centerX, bottomY(0))
+        ..cubicTo(centerX, bottomY(0),
+            centerX, bottomY(0),
+            centerX, bottomY(0))
+        ..cubicTo(centerX, bottomY(0),
+            centerX, bottomY(0),
+            centerX, bottomY(0))
+        ..lineTo(centerX, bottomY(0))
+        ..cubicTo(centerX, bottomY(0),
+            centerX, bottomY(0),
+            centerX, bottomY(0))
+        ..lineTo(centerX, bottomY(0))
+        ..cubicTo(leftX(0.68440646), bottomY(0),
+            leftX(-0.00000004), bottomY(0.68440646),
+            leftX(0), bottomY(1.52866471))
+        ..cubicTo(leftX(0), bottomY(1.52866471),
+            leftX(0), bottomY(1.52866495),
+            leftX(0), bottomY(1.52866495))
+        ..cubicTo(leftX(0), bottomY(1.52866471),
+            leftX(0), bottomY(1.52866471),
+            leftX(0), bottomY(1.52866471))
+        ..lineTo(leftX(0), centerY)
+        ..cubicTo(leftX(0), topY(1.52866483),
+            leftX(0), topY(1.52866483),
+            leftX(0), topY(1.52866483))
+        ..lineTo(leftX(0), topY(1.52866471))
+        ..cubicTo(leftX(0.00000007), topY(0.68440652),
+            leftX(0.68440658), topY(-0.00000001),
+            centerX, topMidY(0))
+        ..cubicTo(centerX, topMidY(0),
+            centerX, topMidY(0),
+            centerX, topMidY(0))
+        ..lineTo(centerX, topMidY(0))
+        ..close();
+    }
+
+    Path roundedRect3b () {
+      return Path()
```

Comment:
We don't ever mutate the paths right? This could be a static final, which will save a tiny bit of resources and still be lazily initialized.

Reply:
So this a subclassed function unfortunately (as it needs to access those variables declared above these functions) so it cannot be made static or final.